contracts/core/Allo.sol

```diff
@@ -1,79 +1,368 @@
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
+    function createPool(
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address payable _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) external payable returns (uint256 poolId) {
+        return _createPool(_identityId, _allocationStrategy, _distributionStrategy, _token, _amount, _metadata);
+    }
+
+    /// @notice Creates a new pool
+    /// @param _identityId The identityId of the pool creator in the registry
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
+    function _createPool(
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address payable _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) internal returns (uint256 poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(_identityId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        if (usedStrategies[_allocationStrategy] || usedStrategies[_distributionStrategy]) {
+            revert STRATEGY_ALREADY_USED();
+        }
+
+        usedStrategies[_allocationStrategy] = true;
+        usedStrategies[_distributionStrategy] = true;
+
+        Pool memory pool = Pool({
+            identityId: _identityId,
+            allocationStrategy: IAllocationStrategy(_allocationStrategy),
+            distributionStrategy: IDistributionStrategy(_distributionStrategy),
+            metadata: _metadata
+        });
+
+        poolId = ++_poolIndex;
+        pools[poolId] = pool;
+
+        if (_amount > 0) {
+            _fundPool(_token, _amount, poolId, address(pool.distributionStrategy));
+        }
+
+        emit PoolCreated(poolId, _identityId, _allocationStrategy, _distributionStrategy, _token, _amount, _metadata);
+    }
+
+    /// @notice passes _data through to the allocation strategy for that pool
+    /// @notice returns the applicationId from the allocation strategy
+    /// @param _poolId id of the pool
+    /// @param _data encoded data unique to the allocation strategy for that pool
     function applyToPool(uint256 _poolId, bytes memory _data) external payable returns (uint256) {
-        // Implement the function here
+        IAllocationStrategy allocationStrategy = pools[_poolId].allocationStrategy;
+
+        return allocationStrategy.applyToPool(_data, msg.sender);
```

Comment:
Can make this one line to match `distribute` and `allocation`.

Reply:
fixed via 1fffea3